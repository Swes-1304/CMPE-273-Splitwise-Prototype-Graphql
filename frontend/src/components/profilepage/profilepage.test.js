import React from 'react';
import { render, cleanup } from '@testing-library/react';
import TestRenderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import Profilepage from './profile_page';
import Navheader from '../navbar/navbar';

const component = TestRenderer.create(
  <MemoryRouter>
    <Profilepage>
      <Navheader />
    </Profilepage>{' '}
  </MemoryRouter>
);

// eslint-disable-next-line no-undef
afterEach(cleanup);

it('renders', async () => {
  expect(component.toJSON()).toMatchSnapshot();
});

test('look for save butoon', async () => {
  const { getByTestId } = render(
    <MemoryRouter>
      <Profilepage>
        <Navheader />
      </Profilepage>{' '}
    </MemoryRouter>
  );
  expect(getByTestId('Saveupdates')).toHaveTextContent('Save');
});
