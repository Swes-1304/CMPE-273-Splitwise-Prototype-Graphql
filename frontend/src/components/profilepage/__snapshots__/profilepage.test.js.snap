// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
Array [
  <div>
    <div>
      <nav
        className="navbar-default navbar navbar-expand navbar-light"
      >
        <a
          className="navbar-brand"
          classname="Navbar-Brand"
          href="/"
          variant="light"
        >
          <img
            alt="icon"
            className="icon"
            src="/Group_Photos/Splitwiseicon.png"
          />
          <span
            className="NavItem"
          >
            Splitwise
          </span>
        </a>
        <div
          aria-expanded={null}
          className="justify-content-end navbar-collapse collapse"
        >
          <div
            className="navbar-nav"
            onKeyDown={[Function]}
          >
            <ul
              className="nav navbar-nav navbar-right"
            >
              <li>
                <button
                  className="login-default btn btn-primary"
                  disabled={false}
                  type="button"
                >
                  <a
                    href="/login"
                    onClick={[Function]}
                  >
                    Login 
                  </a>
                </button>
                or
                 
                <button
                  className="Signup-default btn btn-primary"
                  disabled={false}
                  type="button"
                >
                  <a
                    href="/signup"
                    onClick={[Function]}
                  >
                    Sign Up
                  </a>
                </button>
              </li>
            </ul>
          </div>
        </div>
      </nav>
    </div>
    <div
      className="profilepage-block"
    >
      <h2>
         Your account 
      </h2>
      <section>
        <div
          className="avatar-div"
        >
          <img
            alt="profile pic"
            className="avatar1"
            src="/Profile_photos/null"
          />
          <br />
          <label
            htmlFor="profile_avatar"
          >
            Change your avatar 
            <br />
            <input
              id="profile_avatar"
              name="profile_avatar"
              onChange={[Function]}
              type="file"
            />
          </label>
        </div>
      </section>
      <section>
        <form
          className="profileform"
          id="profileform"
        >
          <section
            className="center-block"
          >
            <div
              className="basic_div"
            >
              <label
                htmlFor="username"
              >
                Your name 
                <br />
                <input
                  defaultValue=""
                  id="username"
                  name="username"
                  onChange={[Function]}
                  type="text"
                />
              </label>
              <br />
              <br />
              <label
                htmlFor="email"
              >
                Your email address 
                <br />
                <input
                  defaultValue=""
                  id="email"
                  name="email"
                  onChange={[Function]}
                  type="email"
                />
              </label>
              <br />
              <br />
              <label
                htmlFor="phonenumber"
              >
                Your phone number 
                <br />
                <input
                  defaultValue=""
                  id="phonenumber"
                  name="phonenumber"
                  onChange={[Function]}
                  type="text"
                />
              </label>
            </div>
          </section>
          <section
            className="right-block"
          >
            <div
              className="default_div"
            >
              <div
                className="form-group"
              >
                <label
                  className="form-label"
                  htmlFor="defaultcurrency"
                >
                  Your default currency
                </label>
                <select
                  className="form-control"
                  id="defaultcurrency"
                  onChange={[Function]}
                  placeholder=""
                  value=""
                >
                  <option
                    value="BHD (BD)"
                  >
                    BHD (BD)
                  </option>
                  <option
                    value="CAD (C$)"
                  >
                    CAD (C$)
                  </option>
                  <option
                    value="EUR (€)"
                  >
                    EUR (€)
                  </option>
                  <option
                    value="GBP (£)"
                  >
                    GBP (£)
                  </option>
                  <option
                    value="KWD (KWD)"
                  >
                    KWD (KWD)
                  </option>
                  <option
                    value="USD ($)"
                  >
                    USD ($)
                  </option>
                </select>
              </div>
              <br />
              <div
                className="form-group"
              >
                <label
                  className="form-label"
                  htmlFor="timezone"
                >
                  TimeZone
                </label>
                <select
                  className="form-control"
                  id="timezone"
                  onChange={[Function]}
                  value=""
                >
                  <option>
                    (GMT -12:00) Eniwetok, Kwajalein
                  </option>
                  <option>
                    (GMT -11:00) Midway Island, Samoa
                  </option>
                  <option>
                    (GMT -10:00) Hawaii
                  </option>
                  <option>
                    (GMT -9:00) Alaska
                  </option>
                  <option>
                    (GMT -8:00) Pacific Time (US & Canada)
                  </option>
                  <option>
                    (GMT -7:00) Mountain Time (US & Canada)
                  </option>
                  <option>
                    (GMT -6:00) Central Time (US & Canada), Mexico City
                  </option>
                  <option>
                    (GMT -5:00) Eastern Time (US & Canada), Bogota, Lima
                  </option>
                  <option>
                    (GMT -4:00) Atlantic Time (Canada), Caracas, La Paz
                  </option>
                  <option>
                    (GMT -3:00) Brazil, Buenos Aires, Georgetown
                  </option>
                  <option>
                    (GMT -2:00) Mid-Atlantic
                  </option>
                  <option>
                    (GMT -1:00) Azores, Cape Verde Islands
                  </option>
                  <option>
                    (GMT) Western Europe Time, London, Lisbon, Casablanca
                  </option>
                  <option>
                    (GMT +1:00) Brussels, Copenhagen, Madrid, Paris
                  </option>
                  <option>
                    (GMT +2:00) Kaliningrad, South Africa
                  </option>
                  <option>
                    (GMT +3:00) Baghdad, Riyadh, Moscow, St. Petersburg
                  </option>
                  <option>
                    (GMT +4:00) Abu Dhabi, Muscat, Baku, Tbilisi
                  </option>
                  <option>
                    (GMT +5:30) Bombay, Calcutta, Madras, New Delhi
                  </option>
                  <option>
                    (GMT +6:00) Almaty, Dhaka, Colombo
                  </option>
                  <option>
                    (GMT +7:00) Bangkok, Hanoi, Jakarta
                  </option>
                  <option>
                    (GMT +8:00) Beijing, Perth, Singapore, Hong Kong
                  </option>
                  <option>
                    (GMT +9:00) Tokyo, Seoul, Osaka, Sapporo, Yakutsk
                  </option>
                  <option>
                    (GMT +10:00) Eastern Australia, Guam, Vladivostok
                  </option>
                  <option>
                    (GMT +11:00) Magadan, Solomon Islands, New Caledonia
                  </option>
                  <option>
                    (GMT +12:00) Auckland, Wellington, Fiji, Kamchatka
                  </option>
                </select>
              </div>
              <br />
              <div
                className="form-group"
              >
                <label
                  className="form-label"
                  htmlFor="language"
                >
                  Language
                </label>
                <select
                  className="form-control"
                  id="language"
                  onChange={[Function]}
                  value=""
                >
                  <option>
                    English
                  </option>
                  <option>
                    Deutsch
                  </option>
                  <option>
                    Italiano
                  </option>
                  <option>
                    Nederlands
                  </option>
                  <option>
                    Svenska
                  </option>
                </select>
              </div>
            </div>
            <div
              className="savebtn"
              data-testid="Saveupdates"
            >
              <button
                className="Signup-default btn btn-primary"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Save
              </button>
            </div>
          </section>
        </form>
      </section>
    </div>
  </div>,
  " ",
]
`;
